petabvis/bar_plot.py

```diff
@@ -82,39 +102,39 @@ def get_bars_df(self, bar_rows):
             df: A dataframe with information relevant
                 for plotting a bar (x, y, sd, etc.)
         """
-        bar_rows = [bar_row for bar_row in bar_rows if
-                    bar_row.dataset_id not in self.disabled_rows]
-
-        x = range(len(bar_rows))
-        tick_pos = range(len(bar_rows))
-        y = [bar.y_data for bar in bar_rows]
-        names = [bar.legend_name for bar in bar_rows]
-        sd = [bar.sd for bar in bar_rows]
-        sem = [bar.sem for bar in bar_rows]
-        noise = [bar.provided_noise for bar in bar_rows]
-        is_simulation = [bar.is_simulation for bar in bar_rows]
-
-        df = pd.DataFrame(
-            list(zip(x, y, names, sd, sem, noise, is_simulation, tick_pos)),
-            columns=["x", "y", "name", "sd", "sem", "provided_noise",
-                     "is_simulation", "tick_pos"])
+        df = self.generate_overview_df()
+        df["x"] = range(len(df.index))
+        df["tick_pos"] = range(len(df.index))
+
+        names = list(df["name"].unique())
+        for i, name in enumerate(names):
+            # set measurement and simulation bars to same x based on name
+            index = df[df["name"] == name].index
+            df.loc[index, "x"] = i - \
+                np.linspace(start=0, stop=self.bar_width,
+                            num=len(df[df["name"] == name].index))
+            df.loc[index, "tick_pos"] = i - self.bar_width/2
```

Comment:
```suggestion
        x = list(range(len(df.index)))
        tick_pos = list(range(len(df.index)))

        for i_name, (name, name_df) in enumerate(df.groupby('name')):
            for i_row in name_df.index:
                x[i_row] = i_name - np.linspace(start=0, stop=self.bar_width, num=len(name_df.index))
                tick_pos[i_row] = i_name - self.bar_width/2

        df.assign(x=x, tick_pos=tick_pos)
```

In general, I think modifying single values of a `pd.DataFrame` one-at-a-time can be very slow, and it's much faster to generate e.g. some sequence and then assign it to the appropriate column when ready.

Might be good to see if this refactor could speed up code elsewhere (would be good to check if this does speed things up first, if you're already familiar with timing Python code).

Reply:
For example, in the table views, if rows are updated individually, this may be why there are some speed issues there

Reply:
I just checked and df.assign seems to be faster. 
However, the speed issue when displaying a large data table can not be caused by this because the data table is not modified when displaying it